ddtrace/contrib/celery/signals.py

```diff
@@ -30,6 +30,11 @@ def trace_prerun(*args, **kwargs):
     # propagate the `Span` in the current task Context
     service = config.celery['worker_service_name']
     span = pin.tracer.trace(c.WORKER_ROOT_SPAN, service=service, resource=task.name, span_type=SpanTypes.WORKER)
+    # set analytics sample rate
+    span.set_tag(
```

Comment:
I think we only want to set the tag here if `config.celery.get_analytics_sample_rate()` is not `None`. 🙂

Reply:
tbh I was just borrowing how the other integrations do it, like [`boto`](https://github.com/DataDog/dd-trace-py/blob/69cd010f37e392bd73e7a9040d0b2c1fee393bc1/ddtrace/contrib/boto/patch.py#L101-L104) for instance, which which just naively sets the return of `get_analytics_sample_rate()`, which would be `None` in the case that `analytics_enabled` is not set or set to false.

Happy to change it if setting a tag to `None` is an antipattern

https://github.com/DataDog/dd-trace-py/blob/4653e97f332ae12a8a5bd8e465c127c5b74bb7c2/ddtrace/settings/integration.py#L82-L98

Reply:
Hmm fair enough. However I think it is an antipattern since TMU `None` will get stringified to `'None'` which is meaningless for this tag 😛.

I'm sure it gets handled and dropped somewhere in the backend but I think it's cleaner if we just don't set it. Less data to process and pass around 🙂

Reply:
makes sense, appreciate the clarification, updated.